### PR TITLE
Enable / Update / Delete stratagem config should show command modal

### LINKED ISSUE #1100: Enable / Update / Delete stratagem config should show command modal
Clicking the Enable / Update or Delete buttons within Stratagem config should show the command modal. This way the user can be notified of progress.

In addition, these commands should take a write-lock on the stratagem object, which we can use to disable the button.

### PR DESCRIPTION
**Pull request by johnsonw**
Fixes #1100.

Clicking the Enable / Update or Delete buttons within Stratagem config should show the command modal. This way the user can be notified of progress.

Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

File: iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs [hunk 1]
```diff
@@ -2,43 +2,48 @@
...
                 orders.skip();
             }
             Err(fail_reason) => {
                 orders.send_msg(Msg::OnFetchError(fail_reason)).skip();
             }
         },
         Msg::OnFetchError(fail_reason) => {
+            model.disabled = false;
             log::error!("Fetch error: {:#?}", fail_reason);
             orders.skip();
```
Comment (jgrund) [inline iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs]: You have `orders.skip()` here, which will prevent the button from changing state.

File: iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs [hunk 2]
```diff
@@ -2,43 +2,48 @@
...
-    StratagemUpdated(fetch::FetchObject<iml_wire_types::Command>),
-    OnFetchError(seed::fetch::FailReason<iml_wire_types::Command>),
+    StratagemUpdated(fetch::FetchObject<ActionResponse>),
+    OnFetchError(seed::fetch::FailReason<ActionResponse>),
 }
 
 pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
     match msg {
         Msg::UpdateStratagem => {
             let orders = orders.skip();
```
Comment (jgrund) [inline iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs]: You have orders.skip() here, which will prevent the button from changing state.

File: iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs [hunk 3]
```diff
@@ -2,43 +2,48 @@
...
 
+            model.disabled = true;
             if let Some(config_data) = &model.config_data {
                 orders.perform_cmd(update_stratagem(&config_data));
             }
         }
         Msg::StratagemUpdated(fetch_object) => match fetch_object.response() {
             Ok(response) => {
                 log::trace!("Response data: {:#?}", response.data);
+                dispatch_custom_event("show_command_modal", &response.data);
```
Comment (jgrund) [inline iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs]: enable again here?

File: iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs (hunk 2 above)
Comment (johnsonw) [reply iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs]: removed

File: iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs (hunk 3 above)
Comment (johnsonw) [reply iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs]: set model.disabled = false

File: iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs (hunk 1 above)
Comment (johnsonw) [reply iml-wasm-components/iml-stratagem/src/update_stratagem_button.rs]: removed

File: iml-wasm-components/iml-stratagem/src/enable_stratagem_button.rs [hunk 4]
```diff
@@ -32,6 +38,8 @@ pub fn update(msg: Msg, model: &mut Model, orders: &mut impl Orders<Msg>) {
         Msg::StratagemEnabled(fetch_object) => match fetch_object.response() {
```
Comment (jgrund) [inline iml-wasm-components/iml-stratagem/src/enable_stratagem_button.rs]: There is an `orders.skip()` at the start of this fn as well.

**Pull request merged**